MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -1,13 +1,73 @@
 using System;
 using System.Collections.Generic;
-using System.Globalization;
+using System.Collections.ObjectModel;
 using System.Linq;
 using System.Text.RegularExpressions;
 using MediaBrowser.Model.Diagnostics;
 using Microsoft.Extensions.Logging;
 
 namespace MediaBrowser.MediaEncoding.Encoder
 {
+    public class FFmpegVersion
+    {
+        private readonly int _major;
+        private readonly int _minor;
+
+        private const int _unknown = 0;
+
+        public FFmpegVersion(string p1)
+        {
+            var match = Regex.Match(p1, @"(?<major>\d+)\.(?<minor>\d+)");
+
+            if (match.Groups["major"].Success && match.Groups["minor"].Success)
+            {
+                _major = int.Parse(match.Groups["major"].Value);
+                _minor = int.Parse(match.Groups["minor"].Value);
+            }
+            else
+            {
+                _major = _unknown;
+                _minor = _unknown;
+            }
+        }
+
+        public override string ToString()
+        {
+            switch (_major)
+            {
+                case _unknown:
+                    return "Unknown";
+                default:
+                    return string.Format("{0}.{1}",_major,_minor);
```

Comment:
even if you won't use string interpolation pls still keep spaces :)
```suggestion
                    return string.Format("{0}.{1}", _major, _minor);
```

Reply:
Shall change to string interpolation